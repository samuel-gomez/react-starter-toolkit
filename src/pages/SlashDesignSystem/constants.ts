export const TITLE_BAR = 'Slash Design System';
export const TITLE = 'Liste des composants';
export const ROUTE_URL_SLASH = '/';

export const COMPONENTS = [
  {
    id: 'text',
    title: 'Champ texte',
    storybook: {
      text: 'Storybook : Text Input',
      path: 'form-input-text--text',
    },
    designSystem: {
      text: 'Guidelines : Text Input',
      path: 'molecules/form-text/',
    },
    picture: {
      name: 'text.svg',
      alt: 'champ texte',
    },
  },
  {
    id: 'textarea',
    title: 'Champ textarea',
    storybook: {
      text: 'Storybook : Textarea Input',
      path: 'form-input-textarea--textarea',
    },
    designSystem: {
      text: 'Guidelines : Textarea Input',
      path: 'molecules/form-textarea/',
    },
    picture: {
      name: 'textrea.svg',
      alt: 'champ texterea',
    },
  },
  {
    id: 'file',
    title: 'Champ file',
    storybook: {
      text: 'Storybook : File Input',
      path: 'form-input-file--file',
    },
    designSystem: {
      text: 'Guidelines : File Input',
      path: 'molecules/form-file/',
    },
    picture: {
      name: 'file.svg',
      alt: 'champ file',
    },
  },
  {
    id: 'accordion',
    title: 'Accordion',
    storybook: {
      text: 'Storybook : Accordion',
      path: 'collapse--accordion',
    },
    designSystem: {
      text: 'Guidelines : Accordion',
      path: 'organisms/accordion/',
    },
    picture: {
      name: 'accordion.svg',
      alt: 'accordion',
    },
  },
  {
    id: 'modal',
    title: 'Modal',
    storybook: {
      text: 'Storybook : Modal',
      path: 'modal-default--modal-core',
    },
    designSystem: {
      text: 'Guidelines : Modal',
      path: 'organisms/modal/',
    },
    picture: {
      name: 'modal.svg',
      alt: 'modal',
    },
  },
  {
    id: 'alert',
    title: 'Alert',
    storybook: {
      text: 'Storybook : Alert',
      path: 'alert-alert--default',
    },
    designSystem: {
      text: 'Guidelines : Alert',
      path: 'molecules/alert/',
    },
    picture: {
      name: 'alert.svg',
      alt: 'alert',
    },
  },
  {
    id: 'badge',
    title: 'Badge',
    storybook: {
      text: 'Storybook : Badge',
      path: 'badge--simple-badge',
    },
    designSystem: {
      text: 'Guidelines : Badge',
      path: 'atoms/badge/',
    },
    picture: {
      name: 'badge.svg',
      alt: 'badge',
    },
  },
  {
    id: 'popover',
    title: 'Popover',
    storybook: {
      text: 'Storybook : Popover',
      path: 'popover--popover',
    },
    designSystem: {
      text: 'Guidelines : Popover',
      path: 'atoms/popover/',
    },
    picture: {
      name: 'popover.svg',
      alt: 'popover',
    },
  },
  {
    id: 'slider',
    title: 'Slider',
    storybook: {
      text: 'Storybook : Slider',
      path: 'form-input-slider--slider',
    },
    designSystem: {
      text: 'Guidelines : Slider',
      path: 'atoms/slider/',
    },
    picture: {
      name: 'slider.svg',
      alt: 'slider',
    },
  },
  {
    id: 'table',
    title: 'Table',
    storybook: {
      text: 'Storybook : Table',
      path: 'table-table--complex',
    },
    designSystem: {
      text: 'Guidelines : Table',
      path: 'molecules/table/',
    },
    picture: {
      name: 'table.svg',
      alt: 'table',
    },
  },
  {
    id: 'tabs',
    title: 'Tabs',
    storybook: {
      text: 'Storybook : Tabs',
      path: 'tabs--default',
    },
    designSystem: {
      text: 'Guidelines : Tabs',
      path: 'molecules/tabs/',
    },
    picture: {
      name: 'tabs.svg',
      alt: 'tabs',
    },
  },
  {
    id: 'button',
    title: 'Button',
    storybook: {
      text: 'Storybook : Button',
      path: 'button--simple-button',
    },
    designSystem: {
      text: 'Guidelines : Button',
      path: 'atoms/button/',
    },
    picture: {
      name: 'button.svg',
      alt: 'button',
    },
  },
  {
    id: 'menu',
    title: 'Menu',
    storybook: {
      text: 'Storybook : Menu',
      path: 'layout-header-navbar--navbarbase',
    },
    designSystem: {
      text: 'Guidelines : Menu',
      path: 'molecules/navigation/',
    },
    picture: {
      name: 'menu.svg',
      alt: 'menu',
    },
  },
  {
    id: 'list',
    title: 'List',
    designSystem: {
      text: 'Guidelines : List',
      path: 'atoms/list/',
    },
    picture: {
      name: 'list.svg',
      alt: 'list',
    },
  },
  {
    id: 'checkbox',
    title: 'Checkbox',
    storybook: {
      text: 'Storybook : Checkbox',
      path: 'form-input-checkbox--checkbox',
    },
    designSystem: {
      text: 'Guidelines : Checkbox',
      path: 'molecules/form-checkbox/',
    },
    picture: {
      name: 'checkbox.svg',
      alt: 'checkbox',
    },
  },
  {
    id: 'checkbox-card',
    title: 'Checkbox-card',
    storybook: {
      text: 'Storybook : Checkbox-card',
      path: 'form-input-checkbox-card--checkbox-card',
    },
    designSystem: {
      text: 'Guidelines : Checkbox-card',
      path: 'organisms/form-card/',
    },
    picture: {
      name: 'checkbox-card.svg',
      alt: 'checkbox-card',
    },
  },
  {
    id: 'radio-card',
    title: 'Radio-card',
    storybook: {
      text: 'Storybook : Radio-card',
      path: 'form-input-radio-card--cardgroupradio-with-two-radio-cards',
    },
    designSystem: {
      text: 'Guidelines : Radio-card',
      path: 'organisms/form-card/',
    },
    picture: {
      name: 'radio-card.svg',
      alt: 'radio-card',
    },
  },
  {
    id: 'steps',
    title: 'Steps',
    storybook: {
      text: 'Storybook : Steps',
      path: 'form-steps--new-design-steps',
    },
    designSystem: {
      text: 'Guidelines : Steps',
      path: 'molecules/steps/',
    },
    picture: {
      name: 'steps.svg',
      alt: 'steps',
    },
  },
  {
    id: 'number',
    title: 'Number',
    storybook: {
      text: 'Storybook : Number',
      path: 'form-input-number--number',
    },
    designSystem: {
      text: 'Guidelines : Number',
      path: 'molecules/form-text/',
    },
    picture: {
      name: 'number.svg',
      alt: 'number',
    },
  },
  {
    id: 'select-multi',
    title: 'Select-multi',
    storybook: {
      text: 'Storybook : Select-multi',
      path: 'form-input-selectmulti--multiselect',
    },
    picture: {
      name: 'select-multi.svg',
      alt: 'select-multi',
    },
  },
  {
    id: 'pass',
    title: 'Pass',
    storybook: {
      text: 'Storybook : Pass',
      path: 'form-input-pass--pass',
    },
    designSystem: {
      text: 'Guidelines : Pass',
      path: 'molecules/form-pass',
    },
    picture: {
      name: 'pass.svg',
      alt: 'pass',
    },
  },
  {
    id: 'date',
    title: 'Date',
    storybook: {
      text: 'Storybook : Date',
      path: 'form-input-date--dateinputstory',
    },
    designSystem: {
      text: 'Guidelines : Date',
      path: 'molecules/form-datepicker/',
    },
    picture: {
      name: 'date.svg',
      alt: 'date',
    },
  },
  {
    id: 'toggle',
    title: 'Toggle',
    storybook: {
      text: 'Storybook : Toggle',
      path: 'form-input-checkbox--checkboxitem-toggle',
    },
    designSystem: {
      text: 'Guidelines : Toggle',
      path: 'molecules/form-checkbox/',
    },
    picture: {
      name: 'toggle.svg',
      alt: 'toggle',
    },
  },
];
