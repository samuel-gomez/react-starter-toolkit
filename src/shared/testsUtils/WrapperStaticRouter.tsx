/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
