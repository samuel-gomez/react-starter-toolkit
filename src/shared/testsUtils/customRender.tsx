/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
