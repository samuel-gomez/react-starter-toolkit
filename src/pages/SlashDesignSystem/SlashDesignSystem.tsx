import Layout, { TLayout } from 'Layout';
import { ReactNode } from 'react';
import Cards from 'shared/components/Cards';
import { COMPONENTS, TITLE_BAR, TITLE } from './constants';

export type TSlashDesignSystemPage = TLayout & {
  titleBar?: ReactNode;
  title?: ReactNode;
  components?: typeof COMPONENTS;
};

const SlashDesignSystemPage = ({ titleBar = TITLE_BAR, title = TITLE, components = COMPONENTS }: TSlashDesignSystemPage) => (
  <Layout propsTitle={{ title: titleBar }}>
    <h2 className="af-title--content">{title}</h2>
    <Cards items={components} />
  </Layout>
);

export default SlashDesignSystemPage;
