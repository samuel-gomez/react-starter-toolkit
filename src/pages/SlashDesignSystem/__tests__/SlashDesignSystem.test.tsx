import { renderWithWrapperStaticRouter } from 'shared/testsUtils';
import { act } from '@testing-library/react';
import SlashDesignSystem from '../SlashDesignSystem';

describe('<SlashDesignSystem page/>', () => {
  it('Renders SlashDesignSystem page component without crashing', async () => {
    await act(async () => {
      const { asFragment } = renderWithWrapperStaticRouter(<SlashDesignSystem />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
