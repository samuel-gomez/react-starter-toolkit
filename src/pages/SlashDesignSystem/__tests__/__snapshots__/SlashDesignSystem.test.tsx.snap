// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlashDesignSystem page/> Renders SlashDesignSystem page component without crashing 1`] = `<DocumentFragment />`;
